src/SourceBrowser/src/BinLogToSln/Program.cs

```diff
@@ -1,21 +1,148 @@
+        {
+            int score = 0;
+
+            try
+            {
+                // Check for UseForSourceIndex escape hatch
+                if (invocation.ProjectProperties.TryGetValue("UseForSourceIndex", out var useForSourceIndex) &&
+                    bool.TryParse(useForSourceIndex, out var shouldUse) && shouldUse)
+                {
+                    return int.MaxValue; // Highest possible score
+                }
+
+                // Check for IsPlatformNotSupportedAssembly property
+                if (invocation.ProjectProperties.TryGetValue("IsPlatformNotSupportedAssembly", out var isPlatformNotSupported) &&
+                    bool.TryParse(isPlatformNotSupported, out var isNotSupported) && isNotSupported)
+                {
+                    score -= 10000; // Heavy penalty for platform not supported assemblies
+                }
+
+                // Prefer invocations with actual source files
+                var sourceFiles = invocation.Parsed?.SourceFiles;
+                if (sourceFiles.HasValue)
+                {
+                    int totalSourceFiles = sourceFiles.Value.Length;
+                    score += totalSourceFiles * 100; // Base score for having source files
+                    
+                    // Bonus for having more source files (indicates more complete implementation)
+                    if (totalSourceFiles > 10)
+                    {
+                        score += 500;
+                    }
+                }
+
+                // Prefer more specific target frameworks using NuGet.Frameworks
+                if (invocation.ProjectProperties.TryGetValue("TargetFramework", out var targetFramework) &&
+                    !string.IsNullOrEmpty(targetFramework))
+                {
+                    try
+                    {
+                        var framework = NuGetFramework.Parse(targetFramework);
+                        
+                        // Prefer platform-specific frameworks
+                        if (framework.HasPlatform)
+                        {
+                            score += 1000;
+                        }
+                        
+                        // Additional scoring based on framework specificity
+                        if (framework.Platform != null)
+                        {
+                            var platformString = framework.Platform.ToLowerInvariant();
+                            if (platformString.Contains("linux") || platformString.Contains("windows") || platformString.Contains("osx"))
+                            {
+                                score += 500;
+                            }
+                        }
+
+                        // Prefer newer frameworks
+                        if (framework.Version != null)
+                        {
+                            score += (int)(framework.Version.Major * 10 + framework.Version.Minor);
```

Comment:
Make sure we give the right weight to framework version.  Here's the precedence I imagine in order of weight from highest to lowest.
1. UseForSourceIndex
2. Not IsPlatformNotSupportedAssembly
3. Newest TargetFramework version
4. Has a platform
5. More source files